google/resource_compute_target_http_proxy.go

```diff
@@ -28,8 +28,9 @@ func resourceComputeTargetHttpProxy() *schema.Resource {
 			},
 
 			"url_map": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
```

Comment:
I don't think the API would accept the resource name, would it? We have a lot of logic around the codebase (see the fns in field_helpers.go and where they get called) to convert between names and self links so users can supply either one, which would enable the use of this diffsuppress. So if you wanted to add that logic that would totally work (and maybe make some people's experience better), but a much easier solution is to use compareSelfLinkRelativePaths instead.

Reply:
Ah, I just took another glance at this, and it looks like there actually _is_ a difference: `.../compute/beta/` vs `.../compute/v1/...`. I was indeed using the self-link of the URL map and not the resource name alone. So it looks like `compareSelfLinkRelativePaths` looks like the better option here. I'll update to use that.